gitlab_integration/models/gitlab_merge_request.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2025 Akretion (http://www.akretion.com).
+# @author Florian Mounier <florian.mounier@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import re
+
+from odoo import api, fields, models
+
+# Tasks ids are handled as [id] [id2] in the beginning of the merge request title
+TASK_REFS_REGEX = re.compile(r"^(?:\s*Draft:\s*)?(?:\s*\[[0-9,]+\]\s*)+")
+TASK_REF_REGEX = re.compile(r"\[([0-9,]+)\]")
+
+
+def extract_task_ids_from_title(title: str) -> list[int]:
+    """Extract task IDs from the merge request title.
+
+    Args:
+        title (str): The title of the merge request.
+    Returns:
+        list[int]: A list of task IDs.
+    """
+    match = TASK_REFS_REGEX.match(title)
+    if not match:
+        return []
+
+    tasks = TASK_REF_REGEX.findall(match.group(0))
+    return [int(task.replace(",", "")) for task in tasks] if tasks else []
+
+
+class GitlabMergeRequest(models.Model):
+    _name = "gitlab.merge.request"
+    _description = "Gitlab Merge Request"
+
+    name = fields.Char(string="Title", required=True)
+    description = fields.Text()
+    draft = fields.Boolean(default=False)
+    state = fields.Selection(
+        [
+            ("opened", "Opened"),
+            ("closed", "Closed"),
+            ("merged", "Merged"),
+            ("locked", "Locked"),
+        ],
+        required=True,
+        default="opened",
+    )
+    gitlab_iid = fields.Integer(required=True)
+    # Make a model?
+    project_path = fields.Char(required=True)
+    project_namespace = fields.Char()
+    project_name = fields.Char()
+
+    web_url = fields.Char()
+
+    task_ids = fields.Many2many(
+        comodel_name="project.task",
+        string="Related Tasks",
+        help="Tasks associated with this Gitlab Merge Request.",
```

Comment:
```suggestion
        help="ERP Tasks associated with this Gitlab Merge Request.",
```

Not to be confused with gitlab's tasks.